client/src/portal/mod.rs

```diff
@@ -210,6 +210,49 @@ impl Keyring {
         }
     }
 
+    /// Verify the secret associated with a keyring.
+    ///
+    /// This function supports all keyring formats (legacy and latest).
+    ///
+    /// # Arguments
+    ///
+    /// *`name` - The name of the keyring.
+    /// * `secret` - The service key, usually retrieved from the Secrets portal.
+    pub async fn verify_keyring_secret(name: &str, secret: Secret) -> bool {
```

Comment:
I am not sure this approach is right at all. Why would you just verify if it is the right secret without returning the corresponding keyring if the secret is ok? why this can't be done by default? there are various questions that needs answers before going on  with an implementation

Reply:
yeah, I agree. I got the same feeling afterwards and that's why I have marked this as a draft :)